app/src/main/kotlin/com/camerasecuritysystem/client/CSSApplication.kt

```diff
@@ -0,0 +1,15 @@
+package com.camerasecuritysystem.client
+
+import android.annotation.SuppressLint
+import android.app.Application
+import android.content.Context
+
+@SuppressLint("StaticFieldLeak")
+object CSSApplication : Application() {
+    private lateinit var _context: Context
+    var context: Context
+        get() = _context
+        set(value) {
+            _context = value
+        }
+}
```

Comment:
Won't this create a memory leak?

Reply:
Is it not possible to give the object that needs that context the context through a constructor or some kind?

Reply:
> Won't this create a memory leak?

No, since only the global application context is saved ([see here](https://github.com/TimGels/Camera-Security-System-Client/blob/footage-response-all/app/src/main/kotlin/com/camerasecuritysystem/client/MainActivity.kt#L42)). Since thehe context is a reference to the global application context and not the specifc context of a single activity, there shouldn't be a leak since the applicationcontext stays alive for the duration of the app process anyway, and won't prevent any activities to clean up.

Source: https://android-developers.googleblog.com/2009/01/avoiding-memory-leaks.html. They recommend to use only the application context at then end of the article.

> Is it not possible to give the object that needs that context the context through a constructor or some kind?

Not for the footage handler, since it is essentially a static class with static functions.

Reply:
As long as it only stores the application context it should be fine.